datadotworld/client.py

```diff
@@ -47,34 +48,45 @@ class Results:
         def __init__(self, raw):
             self.raw = raw
 
-        def asString(self):
+        def __unicode__(self):
+            return self.as_string()
+
+        def __repr__(self):
+            return "{0}\n...".format(self.as_string()[:250])
+
+        def as_string(self):
             return self.raw
 
-        def asStream(self):
+        def as_stream(self):
             return StringIO(self.raw)
 
-        def asDf(self):
+        def as_dataframe(self):
             try:
                 import pandas as pd
             except ImportError:
                 raise RuntimeError("You need to have pandas installed to use .asDf()")
             else:
-                return pd.read_csv(self.asStream())
+                return pd.read_csv(self.as_stream())
 
-        def asCsv(self):
-            return csv.reader(self.asStream())
+        def as_csv(self):
+            # TODO: support UTF-8 formatted CSV in Python 2.x
+            return csv.reader(self.as_stream())
 
     def query(self, dataset, query, query_type="sql"):
-        url = "{}://{}/{}/{}?query={}".format(self.protocol,
-                                              self.queryHost,
-                                              query_type,
-                                              dataset,
-                                              requests.utils.quote(query))
+        from . import __version__
+        params = {
+            "query": query
+        }
+        url = "{0}://{1}/{2}/{3}".format(self.protocol,
+                                         self.queryHost,
+                                         query_type,
+                                         dataset)
         headers = {
-                'Accept': 'text/csv', 
-                'Authorization': 'Bearer ' + self.token
-                }
-        response = requests.get(url, headers=headers)
+            'User-Agent': 'data.world-py - {0}'.format(__version__),
+            'Accept': 'text/csv',
+            'Authorization': 'Bearer {0}'.format(self.token)
+        }
+        response = requests.get(url, params=params, headers=headers)
```

Comment:
aah - good, I hated string-munging the query params, but the (bad) example I found did it that way, so I went with it...  glad to know this works too